crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -149,6 +152,81 @@ impl Solution {
         self.user_trades().next().is_none()
     }
 
+    pub fn merge(&self, other: &Self) -> Result<Self, MergeError> {
+        // We can only merge solutions from the same solver
+        if self.solver.account().address() != other.solver.account().address() {
+            return Err(MergeError::Incompatible("Solvers"));
+        }
+
+        // Solutions should not settle the same order twice
+        let uids = self
```

Comment:
```rust
        let uids: HashSet<_> = self.user_trades().map(|t| t.order().uid).collect();
        let other_uids: HashSet<_> = self.user_trades().map(|t| t.order().uid).collect();
        if uids.intersection(&other_uids).count() != 0 {
            return Err(MergeError::DuplicateTrade);
        }
```

Reply:
btw, `is_disjoint` returns fast. In any case, collecting the second HashSet looks redundant. `.any(|other_uid| uids.contains(&other_uid))` can still be used on the iterator instead.

Reply:
Or to use `is_disjoint`, which returns boolean. But looks like both `is_disjoint` and `intersection` fully iterate through the shorter collection without returning fast on the first found match.